wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt

```diff
@@ -177,4 +221,115 @@ class EditProfileActivity : BaseMenuActivity() {
+                    takePictureIntent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION or Intent.FLAG_GRANT_READ_URI_PERMISSION)
+                }
+                startActivityForResult(takePictureIntent, REQUEST_CODE_IMAGE)
+            }
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode != RESULT_CANCELED) {
+            when (requestCode) {
+                REQUEST_CODE_IMAGE -> {
+                    if (resultCode == RESULT_OK) {
+                        // picture saved in editProfileViewModel.profilePictureTmpFile
+                        editProfileViewModel.onTmpPictureReadyForEditEvent.call(editProfileViewModel.tmpPictureFile)
+                    }
+                }
+                REQUEST_CODE_URI -> if (resultCode == RESULT_OK && data != null) {
+                    val selectedImage: Uri? = data.data
+                    val filePathColumn = arrayOf(MediaStore.Images.Media.DATA)
+                    if (selectedImage != null) {
+                        val cursor: Cursor? = contentResolver.query(selectedImage,
+                                filePathColumn, null, null, null)
+                        if (cursor != null) {
+                            cursor.moveToFirst()
+                            val columnIndex: Int = cursor.getColumnIndex(filePathColumn[0])
+                            val picturePath: String = cursor.getString(columnIndex)
+                            editProfileViewModel.saveAsProfilePictureTmp(picturePath)
+                            cursor.close()
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun cropProfilePicture() {
+        val tmpPictureFile = editProfileViewModel.tmpPictureFile
+        //TODO: this line is for debugging - show the selected image on the screen
+        dashpayUserAvatar.setImageURI(gerFileUri(tmpPictureFile))
+        editProfileViewModel.saveTmpAsProfilePicture()
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        when (requestCode) {
+            REQUEST_CODE_TAKE_PICTURE_PERMISSION -> {
+                when {
+                    grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED -> takePicture()
+                    else -> takePictureWithPermission()
+                }
+            }
+            REQUEST_CODE_CHOOSE_PICTURE_PERMISSION -> {
+                when {
+                    grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED -> choosePicture()
+                    else -> choosePictureWithPermission()
+                }
+            }
+            else -> super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        }
+    }
+
+    private fun gerFileUri(file: File): Uri {
```

Comment:
I think the `ger...` is a typo?

Reply:
Fixed here:  https://github.com/dashevo/dash-wallet/pull/528/commits/206c8d1fd6aee934a93541071dbfbd893a4c6004